### DISCUSSION TITLE
Installation as System Account fails

### DISCUSSION THREAD

**Discussion by Newscp** [sumatrapdfreader/sumatrapdf#2632] [General]
Hi,
till version 3.3.3 it was possible to install Sumatra using wpkg service to deploy the program.
Now with version 3.4.3, the installation fails. There is no log, advice or output. Even with the -log option there won't be a logfile created.
In my opinion the installation fails, because the deploy service run as system account.
If I install Sumatra by command line (as administrator) with the silent switch and the new -all-users switch, everything works fine.
I tried this on win7 and win10-h2 boxes.

kind regards

**Comment by spec1re** [comment]
Hi, same boat here wpkg + sumatrapdf, may I ask how your uninstall line looks like now?  

`"%PROGRAMFILES%\SumatraPDF\SumatraPDF.exe" -uninstall`

Can't get it to uninstall silent with the new version.

**Reply by Newscp** [reply]
I didn't uninstall the new version yet silently. But I assume 

"%PROGRAMFILES%\SumatraPDF\SumatraPDF.exe" -uninstall -s

In the older version (3.3.3) is used "/S"

**Comment by kjk** [comment]
I don't know what `wpkg` does and how it works.

With 3.4 to install system-wide you must use `-all-users` flag. If `wpkg` doesn't do that then it installs for just the current user.

So the first thing is to make sure that `wpkg` runs the installer with `-all-users`

If the log file isn't created it suggest permissions issue.

**Reply by GitHubRulesOK** [reply]
@kjk 
-s is mentioned in docs for filename-install.exe
 but -silent is not mentioned for -uninstall ?
perhaps edit

> -uninstall : uninstalls SumatraPDF (add -silent for silently)

**Reply by Newscp** [reply]
[wpkg](https://wpkg.org/SumatraPDF) is a deployment software. It is installad as a system service executing install routines silently when windows is starting up. It doesn't use the administrator environment.
As i mentioned before, the problem could be that it is executed as the system user. The log file won't be created in the admin temp folder, nor somewhere else.

I tried the `-all-users` option, even with ` -s` but it doesn't work.
With the version 3.3.3 it works pretty fine.

**Reply by kjk** [reply]
Does `wpkg` installation work when it's not running as a service?  Or it fails always with wpkg?

The installer wasn't tested for being run from inside a service. And if even creating a log file doesn't work, then I don't see how I could even debug whatever the issue might be.

If someone figures out the issues and provide a fix then I'll apply it but I'm not particularly interested in figuring this out.

**Reply by Newscp** [reply]
No, wpkg is designed to work as a system service. The new Sumatra Version always fails to install. I tried different commands, with no success. I am not able to use runas /user:system to verify

`   <install cmd='"\\Share\Sumatra\SumatraPDF-3.4.3-64-install.exe" -s -all-users' ><exit code="any" /></install>`    
`   <install cmd='"\\Share\Sumatra\SumatraPDF-3.4.3-64-install.exe" -d "%PROGRAMFILES%\SumatraPDF" -s' ><exit code="any" /></install>`    
`   <install cmd='"\\Share\Sumatra\SumatraPDF-3.4.3-64-install.exe" -s' ><exit code="any" /></install>`

With the older versions it worked smoothly. for ex.
`   <install cmd='"\\Share\Sumatra\SumatraPDF-3.3.3-64-install.exe" -d "%PROGRAMFILES%\SumatraPDF" /S' />`

Maybe you can add in the next version the possibility to change the log file path or use %windir%\temp
As there is no security fix i won't roll out 3.4.3 yet.

**Reply by kjk** [reply]
@Newscp Can you try with `-log` with latest pre-release 15013 or later (from https://www.sumatrapdfreader.org/prerelease) and post the log into a a new issue https://github.com/sumatrapdfreader/sumatrapdf/issues

I've changed the path of log file to be in temp directory so maybe it'll help.
